### PR TITLE
Feat: cmd-warmup add option "check" and "evict"

### LINKED ISSUE #4359: cmd/warmup: support `--check` and `--evict` options
**What would you like to be added**:
Refs: https://juicefs.com/docs/cloud/reference/command_reference#options-15

**Why is this needed**:

### PR DESCRIPTION
**Pull request by jiefenghuang**
close #4359

### CODE REVIEW

File: pkg/vfs/fill.go [hunk 1]
```diff
@@ -184,23 +246,85 @@ func (v *VFS) walkDir(ctx meta.Context, inode Ino, todo chan _file) {
...
+	iter.nextSliceIndex++
+	return s
+}
+
+func (iter *sliceIterator) Iterate(handler sliceHandler) error {
+	for iter.hasNext() {
+		s := iter.next()
+		iter.stat.count++
+		iter.stat.bytes += uint64(s.Size)
+		if handler == nil {
```
Comment (SandyXSD) [inline pkg/vfs/fill.go:304]: check it before the `for` loop

File: pkg/vfs/fill.go [hunk 2]
```diff
@@ -34,36 +34,88 @@ type _file struct {
...
 					}
-					_ = v.Meta.Close(ctx, f.ino)
 				}
-				if count != nil {
-					atomic.AddUint64(count, 1)
+
+				// log and skip error
+				err := iter.Iterate(handler)
+				if err != nil {
+					logger.Error(fmt.Errorf("%s error: %w", action, err))
```
Comment (SandyXSD) [inline pkg/vfs/fill.go:110]: `logger.Errorf`

File: pkg/vfs/fill.go [hunk 3]
```diff
@@ -184,23 +246,85 @@ func (v *VFS) walkDir(ctx meta.Context, inode Ino, todo chan _file) {
...
+type sliceIterStat struct {
+	count uint64
+	bytes uint64
+}
+
+type sliceIterator struct {
+	ctx      meta.Context
+	mClient  meta.Meta
+	ino      Ino
+	size     uint64
```
Comment (SandyXSD) [inline pkg/vfs/fill.go:258]: `size` is not needed.

File: pkg/vfs/fill.go [hunk 4]
```diff
@@ -34,36 +34,88 @@ type _file struct {
...
+						_ = v.Meta.Close(ctx, f.ino)
+					}
+				case EvictCache:
+					handler = func(s meta.Slice) error {
+						return v.Store.EvictCache(s.Id, s.Size)
+					}
+				case CheckCache:
+					handler = func(s meta.Slice) error {
+						missBytes, err := v.Store.CheckCache(s.Id, s.Size)
+						if err != nil {
```
Comment (SandyXSD) [inline pkg/vfs/fill.go:97]: for now err can be ignored.

File: cmd/warmup.go [hunk 5]
```diff
@@ -214,11 +249,20 @@ func warmup(ctx *cli.Context) error {
...
+	} else if check {
+		action = vfs.CheckCache
+	}
+
 	background := ctx.Bool("background")
 	start := len(mp)
 	batch := make([]string, 0, batchMax)
 	progress := utils.NewProgress(background)
-	dspin := progress.AddDoubleSpinner("Warming up")
+	dspin := progress.AddDoubleSpinner(string(action))
```
Comment (SandyXSD) [inline cmd/warmup.go:264]: use `action.String()`

File: pkg/vfs/fill.go (hunk 4 above)
Comment (jiefenghuang) [reply pkg/vfs/fill.go:97]: This error handling is kept here, based on the check API.

File: pkg/vfs/fill.go [hunk 6]
```diff
@@ -34,36 +34,88 @@ type _file struct {
...
+			defer wg.Done()
+			for f := range todo {
+				if ctx.Canceled() {
+					return
 				}
-				if err := v.fillInode(ctx, f.ino, f.size, bytes); err != nil {
-					logger.Errorf("Inode %d could be corrupted: %s", f.ino, err)
+
+				if f.ino == 0 {
+					continue
```
[the remark this answers is not in the document]
Comment (jiefenghuang) [reply pkg/vfs/fill.go:73]: if ino == 0 happen, that means all the other progress should be stopped, maybe we should call ctx.Cancel()
``` go
ctx.Cancel()
return
```

[the remark this answers is not in the document]
Comment (SandyXSD) [reply pkg/vfs/fill.go]: Usually `cancel` is cased by user interruption, like Ctrl+C, while `ino == 0` implies internal error.

**Pull request merged**